core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -389,10 +389,14 @@ private boolean innerIndex(Index index) throws IOException {
             } else {
                 created = update(index, versionValue, indexWriter);
             }
-            Translog.Location translogLocation = translog.add(new Translog.Index(index));
 
-            versionMap.putUnderLock(index.uid().bytes(), new VersionValue(updatedVersion, translogLocation));
-            index.setTranslogLocation(translogLocation);
+            if (index.origin() != Operation.Origin.LOCAL_RECOVERY) {
+                final Translog.Location translogLocation = translog.add(new Translog.Index(index));
```

Comment:
The value is rarely used, so if you think a clearer name is needed, the longer name is fine.